src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/TraceLogging/Statics.cs

```diff
@@ -421,10 +422,11 @@ public static IEnumerable<PropertyInfo> GetProperties(Type type)
             return result;
         }
 
+        [return: MaybeNull]
         public static AttributeType GetCustomAttribute<AttributeType>(PropertyInfo propInfo)
             where AttributeType : Attribute?
         {
-            AttributeType result = null!; // TODO-NULLABLE-GENERIC: re-review
+            AttributeType result = null!;
```

Comment:
Do we really need a ! here? If so, it seems like a bug in the compiler because `AttributeType` is constraint to be `Attribute?`

Reply:
`Attribute` will satisfy the type parameter constraint so `AttributeType` may be a non-nullable type.

Should the method signature be the following perhaps?
```
static AttributeType? GetCustomAttribute<AttributeType>(PropertyInfo propInfo)
    where AttributeType : Attribute
```

Reply:
It doesn't matter much as it is an internal class, but yes, it would make sense to annotate it that way, since we already constraint it to a reference type.

Reply:
Changed. Thanks.